Formula/mimalloc.rb

```diff
@@ -0,0 +1,24 @@
+class Mimalloc < Formula
+  desc "Compact general purpose allocator"
+  homepage "https://github.com/microsoft/mimalloc"
+  url "https://github.com/microsoft/mimalloc/archive/refs/tags/v1.7.2.tar.gz"
```

Comment:
Let's add a comment here that says the `2.x` series is in beta, and that this shouldn't be upgraded to that version until it's stable.